src/pathfinding_service/api.py

```diff
@@ -42,11 +41,15 @@
 
 class ApiWithErrorHandler(Api):
     def handle_error(self, e: exceptions.ApiException) -> Response:
-        log.debug("Error while handling request", error=e, details=e.error_details, message=e.msg)
-        return self.make_response(
-            {"errors": e.msg, "error_code": e.error_code, "error_details": e.error_details},
-            e.http_code,
-        )
+        if isinstance(e, exceptions.ApiException):
```

Comment:
Why do we need this check? Following the annotation this should always be true.

Reply:
The annotation should be more generic. We might get any kind of exception in the error handler, AFAICS.